src/app/Main/Body/QueryPage/QueryPage-Container.component.tsx

```diff
@@ -175,6 +175,19 @@ export function QueryPageContainerComponent(): JSX.Element {
                 [keyName]: chooseSelectedFiltersService(selectedOption),
             },
         };
+        subFilters.forEach((subFilter) => {
```

Comment:
Some general hint why this peace of code and your data model is not the best one:
- You use specific information about the data here: "microorganism" and "matrix". This information is also used in different other parts of your code, so adding a new sub filters would become a difficult task.
- You mix the generic data subFilterTrigger with the specific data. Actually subFilterTriggers are bound to specific filters but this dependency is not visible in this peace of code. So the code looks like it is general but it is not. A more clean way would be to check the subfilters specific to a filter separately. But this would need also some refactoring of the model.